File: backend/src/main/java/com/backend/model/Appointment.java
package com.backend.model;

import com.backend.model.enums.AppointmentStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "appointments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "appointment_id")
    private Long appointmentId;

    // Many appointments belong to one patient
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "patient_id", nullable = false)
    private Patient patient;

    @Column(name = "appointment_datetime", nullable = false)
    private LocalDateTime appointmentDateTime;

    @Column(name = "reason_for_visit", length = 500)
    private String reasonForVisit;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private AppointmentStatus status;

    @OneToMany(mappedBy = "appointment")
    private List<Record> records;

}

File: frontend-v1/types/appointment.ts
import { AppointmentStatus } from "./enums/appointment-status";

// =======================
// Appointment Request DTO
// =======================
export interface AppointmentRequest {
  patientId: number;            // int64 -> number
  appointmentDateTime: string;  // ISO date-time string (e.g., 2026-02-15T10:30:00)
  reasonForVisit: string;       // 0 - 500 characters
  status: AppointmentStatus;               // enum/string from backend
}

// =======================
// Appointment Response DTO
// =======================
export interface AppointmentResponse {
  appointmentId: number;        // int64 -> number
  patientId: number;
  patientName: string;
  appointmentDateTime: string;  // ISO date-time string
  reasonForVisit: string;
  status: AppointmentStatus;
}
